core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,37 @@
+package com.cloudera.spark
+
+import scala.io.Source
+
+class CGroupsMemoryHandle(cgroupMemStats: Seq[(String, Long)]) extends MemoryGetter {
+
+  override val namesAndReporting: Seq[(String, PeakReporting)] = cgroupMemStats.map(x => {
+    (x._1, IncrementBytes)
+  })
+
+  override def values(dest: Array[Long], offset: Int): Unit = {
+    var counter = 0
+    cgroupMemStats.foreach(x => {
+      dest(offset + counter) = x._2
+      counter += 1
+    })
+  }
+}
+
+object CGroupsMemoryHandle {
+
+  val cGroupMemoryStatPath = "/sys/fs/cgroup/memory/memory.stat"
+
+  def get(): Option[CGroupsMemoryHandle] = {
+    if (new java.io.File(cGroupMemoryStatPath).exists) {
+      val cgroupMemStats = Source.fromFile(cGroupMemoryStatPath).getLines
+        .map(line => {
+          val words = line.split(" ")
+          words{0} -> words{1}.toLong
```

Comment:
same on style for `map`

also its standard to use parens, not curly braces, for accessing members of an array, so `words(0) -> words(1).toLong`


btw is it worth checking for a NumberFormatException here?  I forget whether there are any protections in the surrounding code at all, or if this would bring down the whole spark executor.

Reply:
Fixed in the latest commit

Reply:
fixed in latest commit